semgrep-core/src/parsing/tree_sitter/Parse_go_tree_sitter.ml

```diff
@@ -65,9 +65,9 @@ let raw_string_literal env tok =
   (* Remove leading and trailing backticks. The grammar guarantees that raw
    * string literals will always have leading and trailing backticks, so this
    * String.sub call should be safe. Let's check just to be sure. *)
-  if not Common.(s =~ "^`\\(.*\\)`$") then
-    failwith "Found unexpected raw string literal without delimiters";
-  let s = Common.matched1 s in
+  if not (String.get s 0 = '`' && String.get s (String.length s - 1) = '`') then
```

Comment:
This is almost exactly what I wrote in #4671 before I was asked in code review to change it to the regex :laughing:

I think it would be a good idea to check that the string has at least length 2, as well. It always should be, but it would be better to fail with an intelligible error message if that's not the case.

It would still be nice to update the tree sitter grammar to parse the delimiters separately, like it does for other string literals. Then we wouldn't have to use this hack at all. I wrote up more details in #3938.

Reply:
Well, that's unfortunate lol. Updated with your length check suggestion. And yes, changing the tree-sitter grammar would definitely be better.